### PR TITLE
make HTTPS and CORS optional features

### PR DESCRIPTION
**Pull request by pavelsavara**
- adds web-server-use-https
- adds web-server-use-cors
Fixes https://github.com/dotnet/xharness/issues/626

### CODE REVIEW

File: src/Microsoft.DotNet.XHarness.CLI/CommandArguments/TestCommandArguments.cs [hunk 1]
```diff
@@ -45,6 +47,12 @@ protected override OptionSet GetCommandOptions()
                     "ignored. Can be used more than once.",
                     v => _classMethodFilters.Add(v)
                 },
+                { "web-server-use-https", "Bind HTTPS port too",
```
Comment (premun) [inline src/Microsoft.DotNet.XHarness.CLI/CommandArguments/TestCommandArguments.cs:50]: Can you please move the WASM only related args to a WASM command arguments?

`TestCommandArguments` is used across iOS and Android at the moment but the switches don't do anything there.

Applies for even the previously added ones

Comment (pavelsavara) [reply src/Microsoft.DotNet.XHarness.CLI/CommandArguments/TestCommandArguments.cs:50]: We are planning to bring the same functionality to Android soon with @mdh1418

Comment (premun) [reply src/Microsoft.DotNet.XHarness.CLI/CommandArguments/TestCommandArguments.cs:50]: Ah, ok. Then leave it as-is. We will soon be refactoring a bit how we define arguments so taht it's easy to assemble the sets (#431)

**Review by mdh1418 — APPROVED** [review]
Looks good to me 👍

**Review by radical — APPROVED** [review]

**Comment by pavelsavara** [comment]
> What are the situations in which we need to use cors and when would we not need cors?
Android or iOS tests are probably not triggering CORS and therefore it's not necessary to enable it.
CORS is necessary for Browser tests because the http client sends CORS headers.

**Pull request merged**